Attachment for react list form

dudelis opened the pull request:
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| New feature?    | yes                               |
| New sample?     | no                               |

## What's in this Pull Request?

The current Pull Request contains the Attachment functionality for New and Edit forms of the React List Form. A user can add/remove attachments.
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

## Guidance

> *Please update this PR information accordingly. We'll use this as part of our release notes in monthly communications.*
> 
> *Please target your PR to `dev` branch. Released documents are in `master` branch.*
> 
> _(DELETE THIS PARAGRAPH AFTER READING)_

VesaJuvonen commented:
Great stuff @dudelis. Would you like to demonstrate the updated web part in our upcoming community calls?

dudelis commented:
Hi @VesaJuvonen, yes, I could do that. There are 2 things I have added, i.e. editing time for DateTime fields and Attachments. So I could show that and maybe explain a bit how I implemented those.
